### PR TITLE
[Draft] Add grid layout support to SortableList component

### PR DESCRIPTION
**Pull request by vivek20Eng**
- Introduce new 'layout' prop to switch between list and grid layouts
- Update SortableList component to handle grid layout
- Modify CSS to support flexible grid formatting
- Update parent component to use new grid layout option
- Enhance responsiveness and adaptability of the component
- Improve overall flexibility for various use cases

### CODE REVIEW

**Comment by vivek20Eng** [comment]
# Feature Request: Grid Layout Support for SortableList Component

## 📌 Current Situation
We are currently using the SortableList component from the @rodrigodagostino/svelte-sortable-list package in our project. While this component works well for horizontal and vertical layouts, we've identified a need for grid layout support to enhance our user interface.

## 💡 Requested Feature
We kindly request the addition of grid layout support to the SortableList component. This feature would allow items to be arranged and dragged within a grid structure, with automatic wrapping to the next line when space is limited.

## 📈 Use Case
 Need to display and allow reordering of items in a grid format. This layout is crucial for efficiently using screen space and providing a more intuitive interface for our users, especially on devices with larger screens.

## 🚧 Current Limitations
1. The component currently only supports horizontal or vertical layouts.
2. Attempting to implement a grid layout using CSS results in unexpected behavior during drag-and-drop operations.
3. Items do not maintain their grid positions when dragged.

## 🛠️ Proposed Solution
We suggest modifying the SortableList component to:
1. Add a new `layout` prop that accepts 'grid' as a value, in addition to 'horizontal' and 'vertical'.
2. Implement logic to handle item positioning and reordering within a grid structure.
3. Ensure that items wrap to the next line when there's insufficient space in the current row.
4. Maintain grid structure integrity during drag-and-drop operations.


## 📝 Code Example
Here's a basic example of how we envision using the grid layout:

```svelte
<script>
import { SortableList } from '@rodrigodagostino/svelte-sortable-list';

let items = [
  { id: 1, content: 'Item 1' },
  { id: 2, content: 'Item 2' },
  // ... more items
];

function handleSort(event) {
  const { oldIndex, newIndex } = event.detail;
  // Handle reordering logic
}
</script>

<SortableList
  {items}
  layout="grid"
  on:sort={handleSort}
>
  {#each items as item (item.id)}
    <div class="grid-item">
      <span class="icon">🔧</span>
      {item.content}
    </div>
  {/each}
</SortableList>

<style>
.grid-item {
  width: 100px;
  height: 100px;
  margin: 5px;
  background-color: #f0f0f0;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.icon {
  margin-bottom: 5px;
}
</style>


## 🛠️ Current Issue

- [ ] When sorting in grid format, items do not align correctly. This issue is being fixed.

**Comment by vivek20Eng** [comment]
grid-layout-support1
grid-layout-support2

File: src/lib/components/SortableList.svelte [hunk 1]
```diff
@@ -435,68 +436,61 @@
 	}
 </script>
 
-{#if items}
```
Comment (rodrigodagostino) [inline src/lib/components/SortableList.svelte:438]: Please, avoid any unnecessary changes to the current code.

**Comment by rodrigodagostino** [comment]
Due to inactivity, I will be closing this one too for the time being.
Feel free to reopen it if you get back to work on it :)
